panel/pane/web_component.py

```diff
@@ -0,0 +1,424 @@
+"""Implementation of a generic WebComponent pane"""
+
+import ast
+import json
+
+from html.parser import HTMLParser
+from typing import Dict, Optional, Set
+
+import lxml.html as LH
+import param
+
+from bokeh.models import ColumnDataSource
+
+from ..models import WebComponent as _BkWebComponent
+from ..util import escape
+from ..viewable import Layoutable
+from ..widgets.base import Widget
+
+
+# Defines how to convert from attribute string value to parameter value
+PARAMETER_TYPE = {
+    param.String: str,
+    param.Integer: int,
+    param.Number: float,
+    param.ObjectSelector: str,
+    param.Parameter: str,
+    param.List: json.loads,
+    param.Dict: json.loads,
+    param.Boolean: lambda value: value == "" or value is None
+}
+
+
+class AttributeParser(HTMLParser):
+    """Used to parse a the WebComponent html string to a dictionary of attribute keys and
+    their values"""
+
+    first_tag = True
+    attr_dict = None
+
+    def handle_starttag(self, tag, attrs):
+        if self.first_tag:
+            for attr in attrs:
+                self.attr_dict[attr[0]] = attr[1]
+            self.first_tag = False
+
+
+class WebComponent(Widget):
```

Comment:
Is there a reason this lives in `./pane` if it subclasses `Widget`?

Reply:
Yes. Try moving it into `./widgets`. For me it gave an error.